subprojects/resources/src/test/groovy/org/gradle/internal/resource/UriTextResourceTest.groovy

```diff
@@ -188,15 +183,15 @@ class UriTextResourceTest extends Specification {
     }
 
     def hasNoContentWhenUsingFileUriAndFileDoesNotExist() {
-        UriTextResource resource = new UriTextResource('<display-name>', fileUri, resolver)
-        assertFalse(resource.exists)
-        assertNull(resource.file)
-        try {
-            resource.text
-            fail()
-        } catch (MissingResourceException e) {
-            assertThat(e.message, equalTo("Could not read <display-name> '$file' as it does not exist." as String))
-        }
+        when:
+        UriTextResource resource = UriTextResource.from('<display-name>', file, resolver)
+
+        then:
+        resource.exists
```

Comment:
this assertion has been inverted, as we are exercising a MissingTextFileResource instance, which returns true for whatever reason